components/script/dom/htmlimageelement.rs

```diff
@@ -375,6 +437,22 @@ impl HTMLImageElement {
         window.add_pending_reflow();
     }
 
+    fn process_image_response_for_environment_change(&self, image: ImageResponse) {
+        let element = self.upcast::<Element>();
+        let is_multipart_x_mixed_replace = HTMLImageElement::check_mutipart_x_mixed_replace(element);
```

Comment:
This check will need to happen as part of the `process_response` method, by obtaining the mime type of the `Metadata` object that is available there.

Reply:
We were trying to check for the mime type as follows -
`
if let Some(metadata) = metadata {
            if let Some(mime) = metadata.content_type {
                let mime_type = String::from("x-mixed-replace");
                match mime.into_inner() {
                    ContentType::Mime(TopLevel::Multipart, SubLevel::Ext(s), _) => {
                        if s == mime_type {
                            self.abort_request(State::Unavailable, ImageRequestPhase::Pending);
                        }
                    },
                    _ => ()
                }
            }
        }
`
ContentType has to be imported which is not possible because common module is private. Also we are unsure what to do in case of abort since we are inside ImageContext scope?

Reply:
We are able to find the mime type correctly. But still stuck on how to abort here ?

Reply:
The best way to abort is to add a field to ImageContext called `aborted`, set it to true, and return early from all of the handlers if aborted is true.

Reply:
You can import ContentType from the hyper::header module: https://doc.servo.org/hyper/header/index.html